### PR TITLE
Matt electron

### PR DESCRIPTION
**Pull request by Jaitnium**
Log out button is styled. Log out button no longer and no longer breaking the grid formatting.

More scss files refactored. Login and log out button are more responsive.

More comments in Login refactored. No longer checks input fields for empty strings.

Fixed issues where Login wasn't checking if electron was being used. Added functionality to Login in case electron wasn't being used:

If not using electron, there is no way to update local creds, so the input fields are disabled. Login will try to use the local kubeconfig file and will route to the app on success. Otherwise the user is prompted to configure their kubeconfig file themselves.

### CODE REVIEW

File: client/src/Login.tsx [hunk 1]
```diff
@@ -17,155 +18,197 @@ function Login() {
...
+  // Input fields
   const [accessKey, setAccessKey] = useState("");
   const [secretKey, setSecretKey] = useState("");
   const [clusterName, setClusterName] = useState("");
   const [regionName, setRegionName] = useState("");
   // Toggling Password visiblity
   const [hideEye, toggleEye] = useState(true);
   const [hidePassword, togglePassword] = useState(true);
-  // Whether to show the loading screen
+  // Toggle loading screen
```
Comment (drewdunne) [inline client/src/Login.tsx:30]: maybe just rename setLoading to setLoadingScreen then no comment needed

File: client/src/Login.tsx [hunk 2]
```diff
@@ -17,155 +18,197 @@ function Login() {
...
-      setSecretKey(data[1]);
-      setClusterName(data[2]);
-      setRegionName(data[3]);
-    })
+    }
+
+    // Handles electron's response for setting config files
+    if (usingElectron) {
+      window.electronAPI.onConfigResp('onConfigResp', (event: any, data: any) => {
+        // console.log("From server:", data)
```
Comment (drewdunne) [inline client/src/Login.tsx:51]: This looks leftover. Is there a reason to put it into our primary codebase? If so, probably explain.

File: client/src/Login.tsx [hunk 3]
```diff
@@ -17,155 +18,197 @@ function Login() {
   // Don't autoload user into app if they navigated from app
   let { state } = useLocation();
   const autoLoad = state === null ? true : false;
-
+  const usingElectron = window.electronAPI === undefined ? false : true;
```
Comment (lejon1220) [inline client/src/Login.tsx:21]: Is this the logic to fix the getConfig bug?

File: client/src/Login.tsx [hunk 4]
```diff
@@ -17,155 +18,197 @@ function Login() {
...
-      setRegionName(data[3]);
-    })
+    }
+
+    // Handles electron's response for setting config files
+    if (usingElectron) {
+      window.electronAPI.onConfigResp('onConfigResp', (event: any, data: any) => {
+        // console.log("From server:", data)
+        setLoading(false);
+        // The fourth datapoint is whether the kube config was set properly
```
Comment (drewdunne) [inline client/src/Login.tsx:53]: Excellent usage of comments. The "4" here is really unclear. This is also a potentially perfect use case for an enum.

File: client/src/Login.tsx [hunk 5]
```diff
@@ -17,155 +18,197 @@ function Login() {
...
+        } else {
+          alert(badCredentials);
+        }
+      })
+    }
+
+    // Handles electron's response for requesting config files
+    if (usingElectron) {
+      window.electronAPI.onSendConfig('onSendConfig', (event: any, data: [string, string, string, string]) => {
+        // console.log("Parsed from local files:", data);
```
Comment (drewdunne) [inline client/src/Login.tsx:66]: Should this be here? If so, consider explaining why.

File: client/src/Login.tsx [hunk 6]
```diff
@@ -17,155 +18,197 @@ function Login() {
...
+        window.electronAPI.unMount();
+      }
     };
   }, [])
 
-  const loginUser = (loadConfig : boolean) => {
+  /**
+   * Validates and transfers the user to the main app if they have
+   * valid credentials.
+   * @param loadConfig if true and using electron, will load any user
```
Comment (drewdunne) [inline client/src/Login.tsx:88]: Nice!!

File: client/src/Login.tsx [hunk 7]
```diff
@@ -17,155 +18,197 @@ function Login() {
...
+   */
+  const loginUser = (loadConfig: boolean) => {
 
+    let loginUser = false;
     fetch('http://localhost:3000/authenticate', {
       method: 'GET'
     })
       .then(response => {
         //console.log('resp:', resp.json());
         //console.log('typeof:', typeof response.json());
```
Comment (drewdunne) [inline client/src/Login.tsx:99]: Should these be left behind? If so, consider adding why.

File: client/src/Login.tsx [hunk 8]
```diff
@@ -17,155 +18,197 @@ function Login() {
...
   // Return true if the given string is empty
-  const emptyString = (input : string ) => {
-    if(input.trim() === '') return true;
-    return false;    
+  const emptyString = (input: string) => {
+    if (input.trim() === '') return true;
+    return false;
   }
 
+  // Invoked when the user clicks the Login, submit button
```
Comment (drewdunne) [inline client/src/Login.tsx:135]: why not just change the variable name to loginButtonPressed. Then there's no need for this comment.

File: client/src/components/home.tsx [hunk 9]
```diff
@@ -16,16 +16,17 @@ const home  = () => {
   const sourceMap = useAppSelector(selectSourceMap)
   const uiElements: Array<JSX.Element> = [];
 
+  // Make sure to overwite clickable elements with 'pointer-events: auto'
```
Comment (drewdunne) [inline client/src/components/home.tsx:19]: good comment

File: client/src/styles/login.scss [hunk 10]
```diff
@@ -13,27 +20,37 @@
...
-}
+// * {
+//   font-family: SF Pro;
+//   font-size: 16px;
+//   font-weight: 700;
+//   line-height: 19px;
+//   letter-spacing: 0em;
+//   text-align: left;
+//   color: #F4F4F4;
+// }
```
Comment (drewdunne) [inline client/src/styles/login.scss:31]: Leftover code?

File: client/src/styles/theme.scss [hunk 11]
```diff
@@ -85,4 +85,4 @@
 //  letter-spacing: 0em;
 //  text-align: left;
 //  color: #F4F4F4;
```
Comment (drewdunne) [inline client/src/styles/theme.scss:87]: Leftover code?

**Review by drewdunne — APPROVED** [review]

**Review by lejon1220 — APPROVED** [review]
Bug fixes look good

**Pull request merged**